enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -126,12 +126,32 @@ type CacheSnapshotOptions struct {
 	WorkspaceFSPath string
 
 	// Labeled map of chunked artifacts backed by blockio.COWStore storage.
-	ChunkedFiles map[string]*blockio.COWStore
+	ChunkedFiles map[string]*ChunkedFile
 }
 
 type UnpackedSnapshot struct {
 	// ChunkedFiles holds any chunked files that were part of the snapshot.
-	ChunkedFiles map[string]*blockio.COWStore
+	ChunkedFiles map[string]*ChunkedFile
+}
+
+type ChunkedFile struct {
+	*blockio.COWStore
+
+	// digests caches any known digests for *non-dirty* chunks, keyed by offset.
+	digests map[int64]string
+}
```

Comment:
I decided to put this in a snaploader-specific struct instead of adding it to COWStore because right now COWStore is pure I/O and doesn't care about digests etc. which seems more specific to caching. But if that feels somewhat arbitrary, I'm open to changing this. I also wasn't sure whether `ConvertFileToCOW` should eagerly compute the digest. We'll need the digest eventually but if the chunks change by the time we add them to cache then the computation would have been wasteful.

Reply:
I don't really see the benefit of ConvertFileToCOW eagerly computing the digests. If anything it seems better that we push it off until after the workflow is finished if computing the digests ends up being slow